catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -126,10 +126,11 @@ public void shutdown() {
         // The JVM will throw an Interrupted Exception.
       } catch (InterruptedException e) {
         LOGGER.warn("Backup of metacards interrupted. Some metacards might not be backed up.");
+        Thread.currentThread().interrupt();
```

Comment:
❗️ @ryeats, @lessarderic   I know it is outside the scope of this PR, but this logic doesn't seem correct to me. awaitTermination() will return false if it timed out and not throw an InterruptedException. Based on the documentation of this method it should thrown IllegalStateException when it times out which means we should be checking the result of awaitTermination and throw that exception if it is false. 

To properly handle InterruptedException in this method since it is not declared to throw it, we should do the await within a loop that catches the InterruptedException, remembers it was interrupted and either attempts to interrupt the thread that is shutting down or again loop back and await for the remaining duration (must be dynamically calculated) and on the outside of this loop, in a finally, reset the interrupted state of the thread if the flag was previously set to true in this catch.

```
boolean interrupted = false;
long end = System.currentTimeMillis() + getTerminationTimeoutSeconds() * 1000L;

try {
  while (true) {
    try {
      final long timeout = end - System.currentTimeMillis();

      if ((timeout <= 0L) || !executor.awaitTermination(timeout, TimeUnits.MILLISECONDS)) {
        throw new IllegalStateException(...);
      } catch (InterruptedException e) {
        interrupted = true;
     }
  }
} finally {
  if (interrupted) {
    Thread.currentThread().interrupt();
  }
}

Reply:
Are you wanting to create a new ticket for this, or should we fix this here?

Reply:
I will let @lessarderic way in on that.

Reply:
The shutdown method is not part of the `PostIngestPlugin` interface, so it's not an official part of the plugin API. It will only be called by blueprint when the bundle is uninstalled, so it really doesn't matter what exception we throw out.

My recommendation would be to simply implement the normal shutdown code (see https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html), remove the comment and not worry about throwing the `IllegalStateException`.